modules/45_carbonprice/functionalForm/postsolve.gms

```diff
@@ -6,92 +6,233 @@
-				p45_taxCO2eq_anchor(t) = p45_taxCO2eq_anchor(t) + pm_taxCO2eq_anchor_iterationdiff(t) ;
-*** then switch to triangle-approximation based on last two iteration data points			
-			else
-*** change in CO2 price through adjustment: new price - old price; the two instances of p45_taxCO2eq_anchor cancel out -> only the difference term
-				pm_taxCO2eq_anchor_iterationdiff_tmp(t) = 
-				                      max(pm_taxCO2eq_anchor_iterationdiff(t) * min(max((cm_budgetCO2from2020 - s45_actualbudgetco2)/(s45_actualbudgetco2 - s45_actualbudgetco2_last),-2),2),-p45_taxCO2eq_anchor(t)/2);
-				p45_taxCO2eq_anchor(t) = p45_taxCO2eq_anchor(t) + 
-				                      max(pm_taxCO2eq_anchor_iterationdiff(t) * min(max((cm_budgetCO2from2020 - s45_actualbudgetco2)/(s45_actualbudgetco2 - s45_actualbudgetco2_last),-2),2),-p45_taxCO2eq_anchor(t)/2);
-			  pm_taxCO2eq_anchor_iterationdiff(t) = pm_taxCO2eq_anchor_iterationdiff_tmp(t);
-			);
-      o45_taxCO2eq_anchor_iterDiff_Itr(iteration) = pm_taxCO2eq_anchor_iterationdiff("2100");
-		else
-			if(s45_actualbudgetco2 > 0 or abs(cm_budgetCO2from2020 - s45_actualbudgetco2) < 2, !! if model was not optimal, or if budget already reached, keep tax constant
-				p45_taxCO2eq_anchor(t) = p45_taxCO2eq_anchor(t);
-				o45_taxCO2eq_anchor_iterDiff_Itr(iteration) = 0;
-			else
-*** if budget has turned negative, reduce CO2 price by 20%
-				pm_taxCO2eq_anchor_iterationdiff(t) = -0.2*p45_taxCO2eq_anchor(t);
-				p45_taxCO2eq_anchor(t) = p45_taxCO2eq_anchor(t) + pm_taxCO2eq_anchor_iterationdiff(t);
-				o45_taxCO2eq_anchor_iterDiff_Itr(iteration) = pm_taxCO2eq_anchor_iterationdiff("2100");
-			);	
-		);
-		display o45_taxCO2eq_anchor_iterDiff_Itr;
-*** If functionalForm is linear, re-adjust global anchor trajectory to go through the point (cm_taxCO2_historicalYr, cm_taxCO2_historical) 
-$ifThen.taxCO2functionalForm2 "%cm_taxCO2_functionalForm%" == "linear"
-p45_taxCO2eq_anchor(t)$(t.val lt 2110) = s45_taxCO2_historical 
-        + (p45_taxCO2eq_anchor("2110") - s45_taxCO2_historical) / (2110 - s45_taxCO2_historicalYr) !! Yearly increase of CO2 price that interpolates between cm_taxCO2_historical in cm_taxCO2_historicalYr and p45_taxCO2eq_anchor in 2110
+*** --------ALGORITHM for cm_iterative_target_adj eq 5 or 9 ----------------------------------------------------------------------------------------
+*** --------A: calculate the new CO2 price path, beginning with the CO2 tax rescale factor----------------------------------------------------------
+*** --------   this step applies for peak budget and end-of-century budget targets -----------------------------------------------------------------
+if((cm_iterative_target_adj eq 5) OR (cm_iterative_target_adj eq 9),
+
+  if(cm_iterative_target_adj eq 9, !! stronger sensitivity of CO2 price adjustment to CO2 budget deviation for peak budget targets
```

Comment:
Tests suggested that adjustment for end-of-century targets should be smaller (as expected since more timesteps are affected). Thus, making the exponent for the adjustment flexible.